app/services/discourse_subscriptions/campaign.rb

```diff
@@ -0,0 +1,175 @@
+# frozen_string_literal: true
+
+module DiscourseSubscriptions
+  class Campaign
+    include DiscourseSubscriptions::Stripe
+
+    def initialize
+      set_api_key # instantiates Stripe API
+    end
+
+    def refresh_data
+      product_ids = Product.all.pluck(:external_id)
+
+      # if a product id is set for the campaign, we only want to return those results.
+      # if it's blank, return them all.
+      campaign_product = SiteSetting.discourse_subscriptions_campaign_product
+      product_ids = product_ids.select { |id| id == campaign_product } if campaign_product.present?
```

Comment:
I wouldn't recommend looping through the entire `product_ids` array when you know you're only looking for one value.

```ruby
if campaign_product.present?
   product_ids = product_ids.include?(campaign_product) ? [campaign_product] : []
end
```
(This would be significantly faster if you used a `Set.new` too, but if there are few products it probably doesn't matter.

Reply:
Max case is maybe 100 products, so I'm betting the performance increase of `Set.new` would be minimal.

Reply:
It's also like a one line change so I would say it's worth it!